src/nodes/Components/MetaMask.js

```diff
@@ -21,174 +23,167 @@ MetaMask.prototype.onAdded = async function() {
+        } else {
+          console.error(error)
+        }
+    }
+  } else {
+    console.log('Please install MetaMask!')
+  }
 }
 
 MetaMask.prototype.onExecute = async function() {
-
-  if(window.web3 && window.web3.eth && typeof window.web3.eth.getAccounts == "function"){
-    window.web3.eth.getAccounts((error,accounts)=>{
-      this.accounts = accounts
-    })
-  }
-  if(this.accounts){
-    this.setOutputData(0,this.accounts[0])
-  }
-
+  this.connectWeb3()
 
   this.setOutputData(1,{
     name:"balance",
-    args:[{name:"address",type:"string"}],
-    function:async (args)=>{
+    function:async ()=>{
       try{
         this.onAction()
-        let currentWeb3 = new Web3(window.web3)
-        let balance = await currentWeb3.eth.getBalance(args.address)
-        return balance
+        console.log(this.accounts[0])
+        this.Web3 = new Web3(window.ethereum)
+        let balance = await this.Web3.eth.getBalance(this.accounts[0])
+        let eth_balance = this.Web3.utils.fromWei(balance)
+        return eth_balance
       }catch(e){
         console.log(e)
       }
     }
   })
+
+
   this.setOutputData(2,{
     name:"sign",
     args:[{name:"message",type:"string"}],
     function:async (args)=>{
-      return new Promise((resolve, reject) => {
+      await new Promise ((resolve, reject) => {
         this.onAction()
-        let currentWeb3 = new Web3(window.web3)
-        window.ethereum.sendAsync({
+        const from = this.accounts[0]
+        window.ethereum.request({
           method: 'personal_sign',
-          params: [args.message, window.ethereum.selectedAddress],
-          from: window.ethereum.selectedAddress,
-        }, (error,result)=>{
-
-          console.log("SEND MM CALLBACK",error,result)
-          if(error&&error.message){
-            global.setSnackbar({msg:error.message})
-            console.log("REJECT",result)
-            reject(error)
-          }else{
-            console.log("RESOLVE",result)
-            resolve(result.result)
-          }
-
+          params: [args.message, from],
+          from: from,
         })
-      });
-    }
-  })
+        .then((result) => {
+          console.dir(result)
+          console.log('PERSONAL SIGNED: ' + JSON.stringify(result))
+          console.log('recovering...')
+
+          const msgParams = { data: args.message, sig: result }
+          console.log(from.length)
+          const recovered = sigUtil.recoverPersonalSignature(msgParams)
+          console.dir({ recovered })
+
+          if (recovered === from ) {
+            console.log('SigUtil Successfully verified signer as ' + from)
+            resolve(window.alert('SigUtil Successfully verified signer as ' + from))
+          } else {
+            console.dir(recovered)
+            console.log('SigUtil Failed to verify signer when comparing ' + recovered.result + ' to ' + from)
+            reject(console.log('Failed, comparing %s to %s', recovered, from))
+          } 
```

Comment:
ah, I see a lot of these logs already exist. We can remove later as a part of linting. They are also very useful for developers, maybe we could start by setting them all to console.debug unless needed